### ISSUE TITLE
Get partition/agency from Realm custom data

### ISSUE THREAD

**Issue by am-MongoDB** [WildAid/o-fish-android#65]
The user's name and agency are now available (SDK Beta 4) as part of the custom user data.

The app should use that data for:
1. The realm partition should be set to the `agency` from the custom data
2. The user's first and last name for displaying
3. The `agency` attribute for all top-level (non-embedded) Realm Objects
4. The recording office in the Report Object

```
the unit tests are here: https://github.com/realm/realm-java/blob/v10/realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserTests.kt#L338
Basically you just access it from User.getCustomData() and User.refreshCustomData()
realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserTests.kt:338
    fun customData_initiallyEmpty() {
```

![image](https://user-images.githubusercontent.com/14197449/84896748-5d01b080-b09c-11ea-8b6f-7cafee4932ee.png)
